cls/TestCoverage/Data/Coverage.cls

```diff
@@ -70,33 +71,45 @@ ClassMethod StoreIntCoverage(pRun As %Integer, pTestPath As %String, pRoutineNam
 		}
 		
 		Set tCoveredLines = tInstance.CoveredLines
-		
-		Set tAvailableMetrics = ..GetAvailableMetrics()
-		Set tPointer = 0
-		While $ListNext(tAvailableMetrics,tPointer,tMetricKey) {
-			If tInstance.Run.Metrics.Find(tMetricKey) {
-				Set tMetrics(tMetricKey) = $Property(tInstance,tMetricKey)
+		if (pType = "INT")
+		{
+			Set tAvailableMetrics = ..GetAvailableMetrics()
+			Set tPointer = 0
+			While $ListNext(tAvailableMetrics,tPointer,tMetricKey) {
+				If tInstance.Run.Metrics.Find(tMetricKey) {
+					Set tMetrics(tMetricKey) = $Property(tInstance,tMetricKey)
+				}
 			}
-		}
-		
-		// ROWSPEC = "LineNumber:%Integer,LineCovered:%Boolean,RtnLine:%Integer,Time:%Numeric,TotalTime:%Numeric"
-		Set tResult = ##class(TestCoverage.Utils).LineByLineMonitorResultFunc(pRoutineName)
-		While tResult.%Next(.tSC) {
-			$$$ThrowOnError(tSC)
-			Set tLineNumber = tResult.%Get("LineNumber")
-			If tResult.%Get("LineCovered") {
-				Set $Bit(tCoveredLines,tLineNumber) = 1
+			// ROWSPEC = "LineNumber:%Integer,LineCovered:%Boolean,RtnLine:%Integer,Time:%Numeric,TotalTime:%Numeric"
+			Set tResult = ##class(TestCoverage.Utils).LineByLineMonitorResultFunc(pName)
+			While tResult.%Next(.tSC) {
+				$$$ThrowOnError(tSC)
+				Set tLineNumber = tResult.%Get("LineNumber")
+				If tResult.%Get("LineCovered") {
+					Set $Bit(tCoveredLines,tLineNumber) = 1
+				}
+				Set tMetricKey = ""
+				For {
+					Set tMetricKey = $Order(tMetrics(tMetricKey),1,tMetric)
+					If (tMetricKey = "") {
+						Quit
+					}
+					Do tMetric.SetAt(tResult.%Get(tMetricKey) + tMetric.GetAt(tLineNumber),tLineNumber)
+				}
 			}
-			Set tMetricKey = ""
-			For {
-				Set tMetricKey = $Order(tMetrics(tMetricKey),1,tMetric)
-				If (tMetricKey = "") {
-					Quit
+			$$$ThrowOnError(tSC)
+		}
+		Else { // If pType = "PY"
+			//^IRIS.TEMP.TestCoveragePy(ClassName) contains the number of covered lines in this class
+			//^IRIS.TEMP.TestCoveragePy(ClassName, i) in increasing order contain the line numbers for the covered lines
+			if $Data(^IRIS.TEMP.TestCoveragePY(pName)) {
```

Comment:
I'd recommend putting this global name in a macro (in TestCoverage.inc)

Reply:
I did this for the occurrences in Objectscript, but for the reference to ^IRIS.TEMP.TestCoveragePY, since there's no macros in embedded python, I just left it as is with a comment.